docs/aws/audit/redshiftmonitoring/rules/redshift_cluster_encrypted_remediation.mdx

```diff
@@ -1,6 +1,218 @@
+           ClusterIdentifier=cluster_identifier,
+           NodeType=node_type,
+           MasterUsername=master_username,
+           MasterUserPassword=master_user_password,
+           DBName=db_name,
+           Encrypted=True
+       )
+       print(f"Creating encrypted cluster {cluster_identifier}.")
+       return response
+
+   # Example usage
+   create_encrypted_redshift_cluster(
+       cluster_identifier='my-encrypted-cluster',
+       node_type='dc2.large',
+       master_username='admin',
+       master_user_password='YourPassword123',
+       db_name='mydatabase'
+   )
+   ```
+
+By following these steps, you can ensure that your Redshift clusters are encrypted, thereby preventing misconfigurations related to unencrypted clusters.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the Redshift dashboard by selecting "Services" from the top menu, then selecting "Redshift" under the "Database" category.
+3. In the Redshift dashboard, select "Clusters" from the left-hand menu. This will display a list of all your Redshift clusters.
+4. For each cluster, check the "Encryption" column. If the value is "Disabled" or "Not enabled", then the Redshift cluster is not encrypted.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. Make sure you have the necessary permissions to access Redshift resources.
+
+2. Use the following AWS CLI command to list all the Redshift clusters:
+
+   ```
+   aws redshift describe-clusters --query 'Clusters[*].[ClusterIdentifier]' --output text
+   ```
+   This command will return the identifiers of all Redshift clusters.
+
+3. For each cluster identifier returned from the previous step, run the following command to get the encryption status:
+
+   ```
+   aws redshift describe-clusters --cluster-identifier <ClusterIdentifier> --query 'Clusters[*].[Encrypted]' --output text
+   ```
+   Replace `<ClusterIdentifier>` with the actual cluster identifier. This command will return whether the cluster is encrypted or not.
+
+4. If the output of the above command is 'false', then the Redshift cluster is not encrypted. If it's 'true', then the cluster is encrypted. Repeat steps 3 and 4 for each cluster identifier to check the encryption status of all Redshift clusters.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3) on your local system. This will allow you to interact with AWS services including Redshift.
+
+```python
+pip install boto3
+aws configure
+```
+
+2. Import the necessary modules and create a Redshift client using your AWS credentials.
+
+```python
+import boto3
+
+client = boto3.client(
+    'redshift',
+    region_name='us-west-2',
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>